pkg/flow/flow_mem_driver.go

```diff
@@ -809,6 +809,25 @@ func (fc *FlowCollector) updateRecord(record interface{}) error {
 				}
 			} else {
 				if link.EndTime > 0 {
+					if current.Direction != nil && *current.Direction == Incoming {
```

Comment:
iiuc, the removal of the peer link should be regardless of direction. The selection of the peer link should then be for the "opposite" direction.